### ISSUE TITLE
Feature Request: Skip if offline/in maintenance when scheduling tasks/scripts

### ISSUE THREAD

**Issue by branford19** [wh1te909/tacticalrmm#497]
When scheduling tasks, scripts and checks (Whether using the bulk schedule tools, under the agent details or via automation policies), it would be nice to have a "skip if offline" option (Similar to Kaseya), which allows you to skip machines that are not currently reporting in/offline. It may also be pertinent to also skip running the associated task/script on machines which are in maintenance mode, if this option were to be selected.

**Comment by wh1te909** [comment]
this is already happening, scripts and commands are fire and forget so if the agent is offline it's not gonna get it anyway
only thing i could add would be to skip online agents in maintenance mode

**Comment by branford19** [comment]
The "fire and forget" option doesn't appear when scheduling a task:
![image](https://user-images.githubusercontent.com/12677339/118078028-a5366980-b3f8-11eb-9fa5-e48d9855406b.png)

I was thinking possibly to be able to have a regular, scheduled task skipped when a machine is offline/in maintenance mode?

**Comment by wh1te909** [comment]
"add automated task" you can think of a web wrapper around the windows task scheduler. 

create a task and then look in task scheduler on the agent and you'll see the task created. obviously if the computer is turned off then the task wont run. 

and if the agent is "offline" meaning the computer is turned on but has no internet, the task will attempt to run and fail. the way it works is agent just stores the task id, and when it's time to run it reaches out to the rmm with the task id. rmm queries database for that id, and sends back to the agent all the info like the script, timeout etc. so without this info the agent cannot run the task and so it will simply return without running.

so basically if an agent is offline its not going to be able to run anything anyway. hopefully that clears up what i meant by fire and forget. same thing with running bulk commands, scripts etc. rmm will just fire and forget to all the agents you selected. if the agent is not responding the rmm doesn't care it will just skip it and move on to the next

so only option i see is adding a "skip if in maintenance mode"

**Comment by branford19** [comment]
Ah, understood - thank you.

A "skip in maintenance mode" would also be useful :)
